### PR TITLE
[navigation] Left navigation collective

### PR DESCRIPTION
**Pull request by SuZhou-Joe**
### Description

The PR is to fix some issues:
- [x] Change category to manage workspace.
- [x] Move visualization to all use case.
- [x] Change the scrollable area of left navigation.
- [x] Move index patterns management to settings and setup.
- [x] Change the top icon behavior.
- [x] Use gap to add margin to icons in bottom container.
- [x] Make new nav work in mobile
- [x] Add descriptions to some of core apps.
- [x] Use a smaller font size to display menus.
- [x] Append navLinks inside second level to custom category if no  entry for the nav group

### Change category to manage workspace
image

### Change the scrollable area of left navigation.
https://github.com/user-attachments/assets/8f305881-09c5-4305-a582-448bfa04f4fd

### Change the top icon behavior.
image

### Add descriptions to some of core apps.
image

### Append navLinks inside second level to custom category if no entry for the nav group
image

## Changelog

- feat: [navigation] Left navigation collective

### Check List

- [x] All tests pass
  - [x] `yarn test:jest`
  - [x] `yarn test:jest_integration`
- [x] New functionality includes testing.
- [x] New functionality has been documented.
- [x] Update [CHANGELOG.md](./../CHANGELOG.md)
- [x] Commits are signed per the DCO using --signoff

### CODE REVIEW

File: src/core/public/chrome/ui/header/collapsible_nav_group_enabled_top.tsx [hunk 1]
```diff
@@ -4,72 +4,77 @@
...
-  const homeLink = useMemo(() => navLinks.find((link) => link.id === 'home'), [navLinks]);
+  const currentWorkspace = useObservable(currentWorkspace$);
 
-  const isOutsideWorkspace = useMemo(
-    () => !visibleUseCases.find((useCase) => useCase.id === currentNavGroup?.id),
-    [currentNavGroup, visibleUseCases]
+  /**
+   * Only workspace of all use case type will have more than 1 visibleUseCases
+   */
+  const isAllUseCaseWorkspace = useMemo(() => visibleUseCases.length > 1 && !!currentWorkspace, [
```
Comment (ruanyl) [inline src/core/public/chrome/ui/header/collapsible_nav_group_enabled_top.tsx:55]: Nit: I feel `useMemo` is a bit overwhelming here as it's simple expression and returns primitive data type.

File: src/core/public/chrome/ui/header/collapsible_nav_group_enabled_top.tsx [hunk 2]
```diff
@@ -4,72 +4,77 @@
...
+    currentWorkspace,
+    visibleUseCases,
+  ]);
+
+  const isInsideSecondLevel = useMemo(
```
Comment (ruanyl) [inline src/core/public/chrome/ui/header/collapsible_nav_group_enabled_top.tsx:60]: The same here for `useMemo`

File: src/core/public/chrome/ui/header/collapsible_nav_group_enabled_top.tsx [hunk 3]
```diff
@@ -4,72 +4,77 @@
...
   );
 
   const shouldShowBackButton = useMemo(() => {
-    if (!currentNavGroup || currentNavGroup.id === ALL_USE_CASE_ID || shouldShrinkNavigation) {
-      return false;
-    }
-
-    // It means user is in a specific type of workspace
-    if (visibleUseCases.length <= 1) {
+    if (shouldShrinkNavigation) {
```
Comment (ruanyl) [inline src/core/public/chrome/ui/header/collapsible_nav_group_enabled_top.tsx:66]: The same here for `useMemo`, could simply a ternary

File: src/core/public/chrome/ui/header/collapsible_nav_group_enabled_top.tsx [hunk 4]
```diff
@@ -4,72 +4,77 @@
...
-    }
-
-    return visibleUseCases.length > 1;
-  }, [visibleUseCases, currentNavGroup, shouldShrinkNavigation, isOutsideWorkspace]);
+    return isInsideSecondLevel;
+  }, [isInsideSecondLevel, shouldShrinkNavigation]);
 
   const shouldShowHomeLink = useMemo(() => {
-    if (!homeLink || shouldShrinkNavigation) return false;
+    if (shouldShrinkNavigation) return false;
```
Comment (ruanyl) [inline src/core/public/chrome/ui/header/collapsible_nav_group_enabled_top.tsx:74]: The same here for useMemo, could simply a ternary

**Review by ruanyl — DISMISSED** [review]
Left a few minor comments

**Review by ruanyl — APPROVED** [review]

File: src/core/public/chrome/ui/header/collapsible_nav_group_enabled_top.tsx (hunk 3 above)
Comment (SuZhou-Joe) [reply src/core/public/chrome/ui/header/collapsible_nav_group_enabled_top.tsx]: Thanks, updated.

File: src/core/public/chrome/ui/header/collapsible_nav_group_enabled_top.tsx (hunk 4 above)
Comment (SuZhou-Joe) [reply src/core/public/chrome/ui/header/collapsible_nav_group_enabled_top.tsx]: Thanks, updated.

File: src/core/public/chrome/ui/header/collapsible_nav_group_enabled_top.tsx (hunk 2 above)
Comment (SuZhou-Joe) [reply src/core/public/chrome/ui/header/collapsible_nav_group_enabled_top.tsx]: Thanks, updated.

**Review by ruanyl — APPROVED** [review]

**Review by ananzh — APPROVED** [review]

**Review by ruanyl — APPROVED** [review]

**Pull request merged**